pyro/nn/auto_reg_nn.py

```diff
@@ -130,6 +133,10 @@ def __init__(self, input_dim, hidden_dim, output_dim_multiplier=1, permutation=N
         self.output_dim_multiplier = output_dim_multiplier
         self.num_layers = num_layers
 
+        # Hidden dimension must be not less than the input otherwise it isn't
+        # possible to connect to the outputs correctly
+        assert hidden_dim >= input_dim
```

Comment:
nit: since this is user-facing this should be `raise ValueError("...")`

but with that change this PR should be good to merge once tests pass. thank you for your contribution!